src/smt/seq_regex.cpp

```diff
@@ -616,55 +618,57 @@ namespace smt {
                     to_visit.push_back(e2);
                 }
                 // Mark first visit by adding nullptr to the map
-                re_to_bool.insert(e, nullptr);
+                re_to_accept.insert(e, nullptr);
             }
-            else if (re_to_bool.find(e) == nullptr) {
+            else if (re_to_accept.find(e) == nullptr) {
                 // Second visit: set value
                 STRACE("seq_regex_verbose", tout << "2";);
                 to_visit.pop_back();
                 if (m.is_ite(e, econd, e1, e2)) {
-                    expr* b1 = re_to_bool.find(e1);
-                    expr* b2 = re_to_bool.find(e2);
-                    expr* b = m.mk_ite(econd, b1, b2);
+                    expr* b1 = re_to_accept.find(e1);
+                    expr* b2 = re_to_accept.find(e2);
+                    expr* b = m.is_true(econd) || b1 == b2 ? b1 : m.is_false(econd) ? b2 : m.mk_ite(econd, b1, b2);
                     _temp_bool_owner.push_back(b);
-                    re_to_bool.find(e) = b;
+                    re_to_accept.find(e) = b;
                 }
-                /*
-                else if (re().is_empty(e))
+                else if (re().is_empty(e) || (s_is_longer_than_i && re().is_epsilon(e)))
                 {
-                    re_to_bool.find(e) = m.mk_false();
+                    // s[i..] in [] <==> false, also: s[i..] in () <==> false when |s|>i
+                    re_to_accept.find(e) = m.mk_false();
                 }
+                else if (re().is_full_seq(e) || s_is_longer_than_i && re().is_dot_plus(e))
```

Comment:
[528/845] Building CXX object src/smt/CMakeFiles/smt.dir/seq_regex.cpp.o
/home/vsts/work/1/s/src/smt/seq_regex.cpp:639:68: warning: '&&' within '||' [-Wlogical-op-parentheses]
                else if (re().is_full_seq(e) || s_is_longer_than_i && re().is_dot_plus(e))
                                             ~~ ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
/home/vsts/work/1/s/src/smt/seq_regex.cpp:639:68: note: place parentheses around the '&&' expression to silence this warning
                else if (re().is_full_seq(e) || s_is_longer_than_i && re().is_dot_plus(e))
                                                                   ^

Reply:
fixed, will hold off committing yet, until more fixes done, (actually sometimes I do write the parenthesis, just need to be consistent with that) .